dali/python/nvidia/dali/auto_aug/augmentations.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from nvidia.dali import fn
+from nvidia.dali.auto_aug.core import augmentation
+"""
+This module contains a standard suite of augmentations used by AutoAugment policy for ImageNet,
+RandAugment and TrivialAugmentWide. The augmentations are implemented in terms of DALI operators.
+The automatic augmentation schemes parametrize the operations with a magnitude, which,
+intuitively, states how strong the given operation should be.
+
+Each operation defines a range of magnitudes it can accept. The range is divided into a number
+of bins. Then, which bin should be used for a given operation and sample is defined differently
+by different automatic augmentations.
+
+For TrivialAugment, the magnitude bins are chosen randomly every time for every sample.
+In case of RandAugment, the magnitude bin is a fixed hyper-parameter.
+For AutoAugment, the magnitude is a fixed parameter of a given sub-policy.
+
+For some operations, it makes sense to randomly negate the magnitudes to increase
+the variability in the augmented data. Take `fn.random` as an example. Here, the magnitude is
+an angle of the rotation. Negating the angle switches if the rotation is done
+clock- or counterclockwise.
+
+The `@augmentation` lets you to specify what should the range of magnitudes and if the magnitude
+should be randomly negated. Additionally, the `as_param` helps to separate the computation of
```

Comment:
The AA paper states

```
We discretize the range of magnitudes into 10 values (uniform spacing) so that we can use
a discrete search algorithm to find them.
```
so there, the [0.0, p] is the magnitude range. Overall it is a bit ambigious, but instead of renaming the magnitudes into parameters I'd rather put more emphasis on magnitude and magnitude bins. How anout that?